src/pyelliptic/arithmetic.py

```diff
@@ -1,106 +1,144 @@
+G = (Gx, Gy)
+
+
+def inv(a, n):
+    lm, hm = 1, 0
+    low, high = a % n, n
+    while low > 1:
+        r = high / low
+        nm, new = hm - lm * r, high - low * r
+        lm, low, hm, high = nm, new, lm, low
+    return lm % n
 
-def inv(a,n):
-  lm, hm = 1,0
-  low, high = a%n,n
-  while low > 1:
-    r = high/low
-    nm, new = hm-lm*r, high-low*r
-    lm, low, hm, high = nm, new, lm, low
-  return lm % n
 
 def get_code_string(base):
-   if base == 2: return '01'
-   elif base == 10: return '0123456789'
-   elif base == 16: return "0123456789abcdef"
-   elif base == 58: return "123456789ABCDEFGHJKLMNPQRSTUVWXYZabcdefghijkmnopqrstuvwxyz"
-   elif base == 256: return ''.join([chr(x) for x in range(256)])
-   else: raise ValueError("Invalid base!")
-
-def encode(val,base,minlen=0):
-   code_string = get_code_string(base)
-   result = ""   
-   while val > 0:
-      result = code_string[val % base] + result
-      val /= base
-   if len(result) < minlen:
-      result = code_string[0]*(minlen-len(result))+result
-   return result
-
-def decode(string,base):
-   code_string = get_code_string(base)
-   result = 0
-   if base == 16: string = string.lower()
-   while len(string) > 0:
-      result *= base
-      result += code_string.find(string[0])
-      string = string[1:]
-   return result
-
-def changebase(string,frm,to,minlen=0):
-   return encode(decode(string,frm),to,minlen)
-
-def base10_add(a,b):
-  if a == None: return b[0],b[1]
-  if b == None: return a[0],a[1]
-  if a[0] == b[0]: 
-    if a[1] == b[1]: return base10_double(a[0],a[1])
-    else: return None
-  m = ((b[1]-a[1]) * inv(b[0]-a[0],P)) % P
-  x = (m*m-a[0]-b[0]) % P
-  y = (m*(a[0]-x)-a[1]) % P
-  return (x,y)
-  
+    if base == 2:
+        return '01'
+    elif base == 10:
+        return '0123456789'
+    elif base == 16:
+        return "0123456789abcdef"
+    elif base == 58:
+        return "123456789ABCDEFGHJKLMNPQRSTUVWXYZabcdefghijkmnopqrstuvwxyz"
+    elif base == 256:
+        return ''.join([chr(x) for x in range(256)])
+    else:
+        raise ValueError("Invalid base!")
+
+
+def encode(val, base, minlen=0):
+    code_string = get_code_string(base)
+    result = ""
+    while val > 0:
+        result = code_string[val % base] + result
+        val /= base
+    if len(result) < minlen:
+        result = code_string[0] * (minlen - len(result)) + result
+    return result
+
+
+def decode(string, base):
+    code_string = get_code_string(base)
+    result = 0
+    if base == 16:
+        string = string.lower()
+    while string:
+        result *= base
+        result += code_string.find(string[0])
+        string = string[1:]
+    return result
+
+
+def changebase(string, frm, to, minlen=0):
+    return encode(decode(string, frm), to, minlen)
+
+
+def base10_add(a, b):
+    if a is None:
+        return b[0], b[1]
+    if b is None:
+        return a[0], a[1]
+    if a[0] == b[0]:
+        if a[1] == b[1]:
+            return base10_double(a[0], a[1])
```

Comment:
@PeterSurda this looks like unused code
```
>>> import arithmetic
>>> arithmetic.base10_double(0, 1)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: base10_double() takes exactly 1 argument (2 given)
>>> arithmetic.base10_add([0, 1], [0, 1])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "arithmetic.py", line 53, in base10_add
    if a[1] == b[1]: return base10_double(a[0],a[1])
TypeError: base10_double() takes exactly 1 argument (2 given)
```

Reply:
Try an list of two items:
```
>>> arithmetic.base10_double([0, 1])
(0L, 115792089237316195423570985008687907853269984665640564039457584007908834671662L)
```
Although the function is unused (except by arithmetic.base10_add()) there are other functions in there that are not used as well. While removing cruft does fall under linting, I'd argue that applies to our cruft and that we should be using the pyelliptic library rather than copy/pasting it into our code and hacking it.

Perhaps we could open an issue to deal with this separately? Updates to upstream would be easier to manage if we add a dependency or left the code as-is to allow copy/paste from upstream. Having to copy/paste from upstream when we've hacked the files makes updating harder.

If you read this and still want the unused code removing, I will do.

Reply:
It looks like this bug is in upstream since day 1.

Reply:
Upstream is abandoned, sadly. We have to carry the flag.